application_service/app_service.mak

```diff
@@ -62,82 +66,82 @@ clean:
 	@echo "removing executable file"
 	rm -rf $(EXE_DIR)/app_service.exe
 
-hello_world.exe: hello_world.cc
-	@echo "hello_world.cc"
-	$(CC) $(CFLAGS) -o $(O)/hello_world.exe $(S)/hello_world.cc
+hello_world.exe: $(S)/hello_world.cc
+	@echo "compiling $<"
+	$(CC) $(CFLAGS) -o $(@D)/$@ -c $<
 
-send_request.exe: $(O)/send_request.o $(O)/certifier.pb.o $(O)/certifier.o $(O)/support.o $(O)/application_enclave.o $(O)/simulated_enclave.o $(O)/certifier_proofs.o
-	@echo "send_request.exe"
-	$(LINK) -o $(EXE_DIR)/send_request.exe $(O)/send_request.o $(O)/certifier.pb.o \
-	$(O)/certifier.o $(O)/certifier_proofs.o $(O)/support.o $(O)/application_enclave.o $(O)/simulated_enclave.o $(LDFLAGS)
+$(EXE_DIR)/send_request.exe: $(send_req_dobj)
+	@echo "linking executable $@"
+	$(LINK) $(send_req_dobj) $(LDFLAGS) -o $(@D)/$@
 
 $(O)/send_request.o: $(S)/send_request.cc
-	@echo "send_request.cc"
-	$(CC) $(CFLAGS) -c -o $(O)/send_request.o $(S)/send_request.cc
+	@echo "\ncompiling $<"
+	$(CC) $(CFLAGS) -o $(@D)/$@ -c $<
 
-app_service.exe: $(dobj) 
-	@echo "linking executable files"
-	$(LINK) -o $(EXE_DIR)/app_service.exe $(dobj) $(LDFLAGS)
+$(EXE_DIR)/app_service.exe: $(dobj)
+	@echo "linking executable $@"
+	$(LINK) $(dobj) $(LDFLAGS) -o $(@D)/$@
 
-test_user.exe: $(user_dobj) 
-	@echo "linking executable files"
-	$(LINK) -o $(EXE_DIR)/test_user.exe $(user_dobj) $(LDFLAGS)
+$(EXE_DIR)/test_user.exe: $(user_dobj)
+	@echo "linking executable $@"
+	$(LINK) $(user_dobj) $(LDFLAGS) -o $(@D)/$@
 
+$(I)/certifier.pb.h: $(US)/certifier.pb.cc
 $(US)/certifier.pb.cc: $(CP)/certifier.proto
-	$(PROTO) --proto_path=$(CP) --cpp_out=$(US) $<
-	mv certifier.pb.h $(I)
+	$(PROTO) --proto_path=$(<D) --cpp_out=$(@D) $<
+	mv $(@D)/certifier.pb.h $(I)
```

Comment:
This is another instance of a pattern of change where protobuf generated source / output and compilation commands are rewritten to use symbols.